service_management/azure/test/integration/service_bus/sb_queue_gb18030_test.rb

```diff
@@ -1,182 +1,196 @@
-  #     msg = Azure::ServiceBus::BrokeredMessage.new("some text", {
-  #         "key" => ("value" + v).encode("UTF-8")
-  #     })
-
-  #     begin
-  #       subject.send_queue_message queue_name, msg
-  #     end
-  #   }
-  # end
-
-  # it 'Read/Write SB Queue Custom Property Value GB18030' do
-  #   GB18030TestStrings.get.each { |k,v|
-  #     msg = Azure::ServiceBus::BrokeredMessage.new("some text", {
-  #         "key" => ("value" + v).encode("GB18030")
-  #     })
-  #     begin
-  #       subject.send_queue_message queue_name, msg
-  #       retrieved = subject.read_delete_queue_message queue_name
-  #       # Do not flunk here; the service appears to let some
-  #       # "bad" strings go through
-  #     rescue Azure::Core::Http::HTTPError => error
-  #       error.status_code.must_equal 500
-  #     end
-  #   }
-  # end
-end
+#-------------------------------------------------------------------------
+# # Copyright (c) Microsoft and contributors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#--------------------------------------------------------------------------
+require "integration/test_helper"
+require "azure/blob/blob_service"
+
+describe Azure::ServiceBus::ServiceBusService do
+  subject { Azure::ServiceBus::ServiceBusService.new }
+  let(:queue_name) { 'test-queue' }
+
+  describe 'Service bus queue GB-18030' do
+    before do
+      VCR.insert_cassette "service_bus/#{name}"
+      subject.create_queue queue_name
+    end
+
+    after do
+      subject.delete_queue queue_name
+      VCR.eject_cassette
+    end
+
+    it 'should not create queue with UTF-8 names' do
+      # Expected results: Failure, because the Queue name can only contain ASCII
+      # characters, per the Service Bus Queue spec.
+      GB18030TestStrings.get.each do |k,v|
+        begin
+          subject.create_queue queue_name + v.encode("UTF-8")
+          flunk "No exception"
+        rescue
+        end
+      end
+    end
+
+    it 'should not create queue with GB18030 names' do
+      # Expected results: Failure, because the Queue name can only contain ASCII
+      # characters, per the Service Bus Queue spec.
+      GB18030TestStrings.get.each do |k,v|
+        begin
+          subject.create_queue queue_name + v.encode("GB18030")
+          flunk "No exception"
+        rescue
+        end
+      end
+    end
+
+    # FIXME VCR does not record cassette when request/response body has unrecognized characters

```

Comment:
Sure I'll open it after completing the PR so that i have the exact line to reference it. :)